### PR TITLE
MF-737: validate biometrics and vitals form inputs to only allow numerals

### PR DESCRIPTION
**Pull request by KorirC**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://app.zeplin.io/project/60d59321e8100b0324762e05/screen/60d5df2840a9c1136401d7c7).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary

-validate biometrics and vitals form inputs to only allow numerals

## Screenshots

MF-737

## Issue

[MF-737](https://issues.openmrs.org/browse/MF-737)

## Other

*None.*

### CODE REVIEW

File: packages/esm-patient-vitals-app/src/vitals/vitals-biometrics-form/vitals-biometrics-input.component.tsx [hunk 1]
```diff
@@ -33,6 +35,14 @@ const VitalsBiometricInput: React.FC<VitalsBiometricInputProps> = ({
   inputIsNormal,
   isTablet,
 }) => {
+  const { t } = useTranslation();
+  const [invalid, setInvalid] = useState<boolean>(false);
+  const check = (value) => {
+    Number(value) || value === '' ? setInvalid(false) : setInvalid(true);
+  };
```
Comment (brandones) [inline packages/esm-patient-vitals-app/src/vitals/vitals-biometrics-form/vitals-biometrics-input.component.tsx:42]:
```suggestion
  function check(value) {
    setInvalid(!(Number(value) || value === '') )
  }
```

**Comment by brandones** [comment]
It looks like somehow we lost the other good work you did? Please change the input type to `number` and fix the `val.type`s to be `number` instead of text. Same as we had before, but with these new validators.

**Comment by KorirC** [comment]
> It looks like somehow we lost the other good work you did? Please change the input type to `number` and fix the `val.type`s to be `number` instead of text. Same as we had before, but with these new validators.

Hi @brandones, concerning this when i change to `type=number` and maintain the new validators, with firefox browser the input fields accepts letters and doesn't throw the validation error. it only validates when input is 0.

File: packages/esm-patient-vitals-app/src/vitals/vitals-biometrics-form/vitals-biometrics-input.component.tsx [hunk 2]
```diff
@@ -33,6 +35,23 @@ const VitalsBiometricInput: React.FC<VitalsBiometricInputProps> = ({
...
 }) => {
+  const { t } = useTranslation();
+  const [invalid, setInvalid] = useState<boolean>(false);
+
+  function check(value) {
+    setInvalid(!(Number(value) || value === ''));
+  }
+  const error = () => {
+    return <FormLabel className={styles.danger}>{t('numericInputError', 'Must be a number')}</FormLabel>;
+  };
```
Comment (brandones) [inline packages/esm-patient-vitals-app/src/vitals/vitals-biometrics-form/vitals-biometrics-input.component.tsx:46]: Just inline this JSX in the return block. Avoid having JSX distributed around your components, when possible.

**Review by brandones — APPROVED** [review]
Looks good, thanks @KorirC !

**Pull request merged**